include/ginkgo/core/log/logger.hpp

```diff
@@ -388,6 +388,18 @@ public:                                                              \
         const uint8 &stopping_id, const bool &set_finalized,
         const Array<stopping_status> *status, const bool &one_changed,
         const bool &all_converged)
+protected:
+    virtual void on_criterion_check_completed(
```

Comment:
Either here, or in the block above for `GKO_LOGGER_REGISTER_EVENT` for `criterion_check_completed`, it would be nice if you document what `implicit_sq_tau` is, so that people who create new solvers/wrappers know what this is about. If you add it to the block above, you could somehow include it as an optional argument by just adding a line without `\param`.